velox/type/tests/DecimalTest.cpp

```diff
@@ -23,6 +23,31 @@ namespace facebook::velox {
 
 namespace {
 
+template <typename T>
+void assertRescaleDouble(double value, const TypePtr& type, T expectedValue) {
+  const auto [precision, scale] = getDecimalPrecisionScale(*type);
+  T actualValue;
+  const auto status =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, actualValue);
+  ASSERT_TRUE(status.ok());
+  ASSERT_EQ(actualValue, expectedValue);
+}
+
+template <typename T>
+void assertRescaleDoubleFail(
+    double value,
+    const TypePtr& type,
+    const std::string& expectedErrorMessage) {
+  const auto [precision, scale] = getDecimalPrecisionScale(*type);
+  T actualValue;
+  const auto status =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, actualValue);
+  ASSERT_TRUE(!status.ok());
+  ASSERT_TRUE(status.message().find(expectedErrorMessage) != std::string::npos)
```

Comment:
We may need to add a macro similar to VELOX_ASSERT_THROW for calls that return Status to avoid copy-pasting this many times.

Reply:
Added `VELOX_ASSERT_ERROR_STATUS` to check an error status. Thanks.